admin/sql/create_tables.sql

```diff
@@ -18,4 +18,22 @@ CREATE TABLE listen (
   raw_data        JSONB
 );
 
+CREATE TABLE tokens (
+     id               SERIAL,
+     user_id          VARCHAR,
+     token            TEXT NOT NULL,
+     api_key          VARCHAR NOT NULL,
+     ts               TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+);
+ALTER TABLE tokens ADD CONSTRAINT tokens_api_key_uniq UNIQUE (api_key);
+
+CREATE TABLE sessions (
+    id        SERIAL,
+    user_id   VARCHAR NOT NULL,
```

Comment:
**Thanks for pointing out! I almost forgot to revert this change.**
Earlier i was trying to keep this column just for storing extra info but afterwards i changed a lot of schema to incorporate api-key checks at every stage which required a lot of table joins to access the **"user"** table.

I'll have to reduce the number of queries for a single batch, to improve performance. Converting columns to foreign keys is part of this task.